src/PointwiseFunctions/Hydro/EquationsOfState/Enthalpy.cpp

```diff
@@ -0,0 +1,451 @@
+  exponential_integral_coefficients.polynomial_coefficients =
+      integral_poly_coeffs;
+  exponential_integral_coefficients.sin_coefficients = integral_sin_coeffs;
+  exponential_integral_coefficients.cos_coefficients = integral_cos_coeffs;
+  double new_constant = initial_condition.second -
+                        evaluate_coefficients(exponential_integral_coefficients,
+                                              initial_condition.first);
+  exponential_integral_coefficients.exponential_external_constant =
+      new_constant;
+  return exponential_integral_coefficients;
+}
+template <typename LowDensityEoS>
+typename Enthalpy<LowDensityEoS>::Coefficients
+Enthalpy<LowDensityEoS>::Coefficients::compute_derivative() {
+  // Slow/Bad?
+  std::vector<double> derivative_poly_coeffs(polynomial_coefficients.size());
+  std::vector<double> derivative_sin_coeffs(sin_coefficients.size());
+  std::vector<double> derivative_cos_coeffs(cos_coefficients.size());
+  Enthalpy::Coefficients derivative_coefficients = *this;
+  // d/dz e^z \sum_i a_i f(z) = e^z \sum_i a_i f_i(z)  + e^z \sum_i a_i f_i'(z)
+  if (has_exponential_prefactor) {
+    ERROR("You shouldn't have to do this!");
+    derivative_poly_coeffs = polynomial_coefficients;
+    derivative_sin_coeffs = sin_coefficients;
+    derivative_cos_coeffs = cos_coefficients;
+    derivative_poly_coeffs[polynomial_coefficients.size() - 1] = 0.0;
+    for (size_t i = 0; i < polynomial_coefficients.size() - 1; i++) {
+      derivative_poly_coeffs[i] +=
+          polynomial_coefficients[i + 1] * static_cast<double>(i + 1);
+    }
+    // Again sum starts from 0
+    for (size_t j = 0; j < sin_coefficients.size(); j++) {
+      derivative_cos_coeffs[j] +=
+          sin_coefficients[j] * (static_cast<double>(j + 1) * trig_scale);
+      derivative_sin_coeffs[j] +=
+          -cos_coefficients[j] * (static_cast<double>(j + 1) * trig_scale);
+    }
+  } else {
+    // The final coefficient will be zero because nothing differentiates to it
+    derivative_poly_coeffs[polynomial_coefficients.size() - 1] = 0.0;
+    for (size_t i = 0; i < polynomial_coefficients.size() - 1; i++) {
+      derivative_poly_coeffs[i] =
+          polynomial_coefficients[i + 1] * static_cast<double>(i + 1);
+    }
+    for (size_t j = 0; j < sin_coefficients.size(); j++) {
+      derivative_cos_coeffs[j] =
+          sin_coefficients[j] * (static_cast<double>(j + 1) * trig_scale);
+      derivative_sin_coeffs[j] =
+          -cos_coefficients[j] * (static_cast<double>(j + 1) * trig_scale);
+    }
+  }
+  derivative_coefficients.polynomial_coefficients = derivative_poly_coeffs;
+  derivative_coefficients.sin_coefficients = derivative_sin_coeffs;
+  derivative_coefficients.cos_coefficients = derivative_cos_coeffs;
+  return derivative_coefficients;
+}
+template <typename LowDensityEoS>
+void Enthalpy<LowDensityEoS>::Coefficients::pup(PUP::er& p) {
+  p | polynomial_coefficients;
```

Comment:
Ah yeah I added `minimum_enthalpy_` to the class as a convenience, I will need to send it to reconstruct the class.  Is this bad practice?  It's not really independent data, just annoying to recompute in `rest_mass_density_from_enthalpy`

Reply:
I think it's totally fine to have a variable for convenience! It might be an issue if it were like 500MB of data, but a single double, or even 100 of them aren't going to make a difference :)